src/components/tables/TableConnected.tsx

```diff
@@ -28,6 +29,9 @@ const mapStateToProps = (state: IReactVaporState, ownProps: ITableOwnProps): ITa
   const perPageState: IPerPageState = tableState && _.findWhere(state.perPageComposite, { id: tableState.perPageId });
   const tableHeaderCellState: ITableHeaderCellState = tableState && state.tableHeaderCells[tableState.tableHeaderCellId];
   const predicateStates: IDropdownSearchState[] = tableState && _.reject(state.dropdownSearch, (dropdownSearch: IDropdownSearchState) => !contains(dropdownSearch.id, ownProps.id)) || [];
+  const datePickerState: IDatePickerState = tableState && _.find(state.datePickers, (datePicker: IDatePickerState) => {
```

Comment:
Unless I'm mistaken, I think you could use `const datePickerState: IDatePickerState = tableState && _.findWhere(state.datePickers, { id: tableState.datePickerId });` here.

Reply:
It doesn't work because children of tableChildComponents have a different id example `react-vapor-table124datepicker` for the table `react-vapor-table124`. I could parse out `datepicker` before `findWhere` and it would work! Cleaner also

Reply:
Oh okay, I wasn't aware of that. Would the following work then?
``_.findWhere(state.datePickers, { id: `${tableState.datePickerId}datepicker` })``
IMHO its cleaner and simpler to add to an existing string, than to parse it.

Reply:
Can you use `getTableChildComponentId(tableState.datePickerId, TableChildComponent.DATEPICKER)` to get the component id?

Reply:
You should remove the curly braces and the return also.

Reply:
What works is this: `const datePickerState: IDatePickerState = tableState && _.findWhere(state.datePickers, { id: `${tableState.datePickerId}-date-range` });`

I think there is a reason why indexOf is used as `-date-range` might not always be there, or it might be something else

Same method is also here : https://github.com/coveo/react-vapor/blob/master/src/components/datePicker/DatePickerDropdownConnected.tsx#L24

Reply:
@ThibodeauJF are you saying that the datepicker you retrieve doesn't always have the same id? if it doesn't that I understand your strategy, otherwise if you always retrieve ``${tableState.datePickerId}-date-range}`` i would consider making a special case in `getTableChildComponentId` for it where the id of TableChildComponent.DATEPICKER will be `${tableId}${TableChildComponent.DATEPICKER}-date-range` . If it's not always that I would add a comment explaining the situation that the id is not always the same (which is a bit weird actually).

Reply:
in otherwise if you can, you should make it so that you can user `_.findWhere(state.datePickers, { id: tableState.datePickerId });` since the only use case of `tableState.datePickerId` seems to be here

Reply:
If you want to add a datePicker with no date range (single date), the Id will look like this: `datepicker-react-vapor-table5-select-date`. Right now both are possible with the feature (props `isLinkedToDateRange` and `datesSelectionBoxes` with single `quickOptions` and `isRange` to false)

Reply:
I would add a comment that explains this